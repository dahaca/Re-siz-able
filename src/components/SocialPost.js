import React from 'react'
import styled from 'styled-components'
import Resizer from '../lib/Resizable/Resizer'

export const PostContainer = styled.div`
  position: relative;
  height: 20rem;
  width: 35rem;
  padding: 6rem 2.5rem 2.5rem 2.5rem;
  background: #fff;
  box-shadow: 0px 5px 10px 0 #d2d2d2;
  border-radius: 10px;
  border-bottom: 15px solid #87cefa;
`

export const Name = styled.div`
  position: absolute;
  top: 2rem;
  left: 2.5rem;
  font-size: 2rem;
  font-weight: 500;
`

export const Close = styled.div`
  position: absolute;
  right: 1rem;
  top: 1rem;
  height: 1.5rem;
  width: 1.5rem;
  border-radius: 50%;
  background: #000;
`

export const Description = styled.div`
  max-width: 30rem;
  margin-bottom: 2rem;
  line-height: 1.7rem;
`

const SocialPost = () => (
  <Resizer
    handles={['bottom', 'right', 'left']}
    minHeight={300}
    maxHeight={600}
    minWidth={300}
  >
    <PostContainer>
      <Name>John Doe</Name>
      <Close />
      <Description>
        A functional component which internally uses the Resizer component.
        Bottom, right and left handles are enabled. Min width is set to 250px,
        min height to 150 and max to 500px.
      </Description>
    </PostContainer>
  </Resizer>
)

export default SocialPost
