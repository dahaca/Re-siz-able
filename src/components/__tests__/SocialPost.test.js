import React from 'react'
import { render } from 'enzyme'
import SocialPost from '../SocialPost'

describe('Test SocialPost', () => {
  it('should render correctly', () => {
    expect(render(<SocialPost />)).toMatchSnapshot()
  })
})
