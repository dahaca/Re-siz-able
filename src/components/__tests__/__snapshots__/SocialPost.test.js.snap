// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SocialPost should render correctly 1`] = `
<div
  class="sc-bwzfXH fbmalI"
>
  <div
    class="sc-htpNat cCdKpA"
  >
    <div
      class="sc-bxivhb cgeqyC"
    >
      John Doe
    </div>
    <div
      class="sc-ifAKCX lgNMH"
    />
    <div
      class="sc-EHOje jCbNlh"
    >
      A functional component which internally uses the Resizer component. Bottom, right and left handles are enabled. Min width is set to 250px, min height to 150 and max to 500px.
    </div>
  </div>
  <div
    class="sc-bdVaJa eOejiP"
  />
  <div
    class="sc-bdVaJa iFZMkA"
  />
  <div
    class="sc-bdVaJa ifPxOL"
  />
</div>
`;
